### ISSUE TITLE
is_subclass_of doesn't reveal properties of an already specialised class-string

### ISSUE THREAD

**Issue by aidantwoods** [vimeo/psalm#1219]
https://getpsalm.org/r/2aa4c9ac7b

```php
<?php
interface Foo {
  public static function Bar() : bool;
};

interface Bar {}

class FooClass implements Foo, Bar {
  public static function Bar() : bool {
    return true;
  }
}


/**
 * @param class-string<Bar> $className
 */
function foo($className) : bool {
  if (is_subclass_of($className, Foo::class, true)) {
    return $className::Bar();
  }
  
  return false;
}
```

Psalm output (using commit 590bea3): 
```
INFO: MixedMethodCall - 20:12 - Cannot call constructor on an unknown class

INFO: MixedReturnStatement - 20:5 - Could not infer a return type

INFO: MixedInferredReturnType - 18:28 - Could not verify return type 'bool' for foo
```

---

I suspect this might be because Psalm can't currently represent non-object intersection types?

e.g. `@param class-string<Bar>&class-string<Foo> $className` gives:  `ERROR: InvalidDocblock [...] Intersection types must all be objects, [...]`, and the slightly preferable `@param class-string<Bar&Foo> $className` gives:  `ERROR: UndefinedClass [...] Class or interface Bar&Foo does not exist`

**Comment by muglug** [comment]
`@param class-string<Bar&Foo> $className` doesn't give the error anymore, but it still doesn't work...
